interfaces/builtin/fwupd.go

```diff
@@ -0,0 +1,229 @@
+  /sys/firmware/efi/esrt/entries/** r,
+
+  # Allow fwupd to access system information
+  /sys/devices/virtual/dmi/id/product_name r,
+  /sys/devices/virtual/dmi/id/sys_vendor r,
+
+  # File accesses
+  # Allow read access for efivarfs filesystem
+  /sys/firmware/efi/efivars/ r,
+  /sys/firmware/efi/efivars/** r,
+
+  # DBus accesses
+  #include <abstractions/dbus-strict>
+  dbus (send)
+      bus=system
+      path=/org/freedesktop/DBus
+      interface=org.freedesktop.DBus
+      member={Request,Release}Name
+      peer=(name=org.freedesktop.DBus),
+
+  dbus (receive, send)
+      bus=system
+      path=/org/freedesktop/DBus
+      interface=org.freedesktop.DBus
+      member=GetConnectionUnixUser
+      peer=(label=unconfined),
+
+  # Allow binding the service to the requested connection name
+  dbus (bind)
+      bus=system
+      name="org.freedesktop.fwupd",
+`)
+
+var fwupdConnectedPlugAppArmor = []byte(`
+# Description: Allow using fwupd service. Reserved because this gives
+#  privileged access to the fwupd service.
+# Usage: reserved
+
+  # DBus accesses
+  #include <abstractions/dbus-strict>
+
+  # Allow access to fwupd service
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.fwupd
+      peer=(label=###SLOT_SECURITY_TAGS###),
+
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.DBus.Properties
+      peer=(label=###SLOT_SECURITY_TAGS###),
+`)
+
+var fwupdConnectedSlotAppArmor = []byte(`
+# Description: Allow firmware update from fwupd service. Reserved because this gives
+#  privileged access to the system.
+# Usage: reserved
+
+  # Allow libfwup to access efivarfs with immutable flag
+  capability linux_immutable,
+
+  # File accesses
+  # Allow write access for efivarfs filesystem
+  /sys/firmware/efi/efivars/** rw,
+
+  # Allow write access for efi firmware updater
+  /boot/efi/EFI/ubuntu/fw/** rw,
+  # Allow access from efivar library
+  @{PROC}/@{pid}/mounts r,
+  /sys/devices/**/partition r,
+  /dev/sd[a-z]* r,
+  # Allow access UEFI firmware platform size
+  /sys/firmware/efi/ r,
+  /sys/firmware/efi/fw_platform_size r,
+
+  # Allow traffic to/from org.freedesktop.DBus for fwupd service
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.DBus.**
+      peer=(label=###PLUG_SECURITY_TAGS###),
+
+  dbus (receive, send)
+      bus=system
+      path=/org/freedesktop/fwupd{,/**}
+      interface=org.freedesktop.DBus.**
+      peer=(label=###PLUG_SECURITY_TAGS###),
+
+  # Allow traffic to/from fwupd interface with any method
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.fwupd
+      peer=(label=###PLUG_SECURITY_TAGS###),
+
+  dbus (receive, send)
+      bus=system
+      path=/org/freedesktop/fwupd{,/**}
+      interface=org.freedesktop.fwupd
+      peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var fwupdPermanentSlotDBus = []byte(`
+<policy user="root">
```

Comment:
Some interface do use them when I checked the current builtin interfaces

$ grep -C 2 "policy user=\"root" interfaces/builtin/*.go
interfaces/builtin/bluez.go-
interfaces/builtin/bluez.go-var bluezPermanentSlotDBus = []byte(`
interfaces/builtin/bluez.go:<policy user="root">
interfaces/builtin/bluez.go-    <allow own="org.bluez"/>
interfaces/builtin/bluez.go-    <allow own="org.bluez.obex"/>
--
interfaces/builtin/fwupd.go-
interfaces/builtin/fwupd.go-var fwupdPermanentSlotDBus = []byte(`
interfaces/builtin/fwupd.go:<policy user="root">
interfaces/builtin/fwupd.go-    <allow own="org.freedesktop.fwupd"/>
interfaces/builtin/fwupd.go-</policy>
--
interfaces/builtin/location_control.go-
interfaces/builtin/location_control.go-var locationControlPermanentSlotDBus = []byte(`
interfaces/builtin/location_control.go:<policy user="root">
interfaces/builtin/location_control.go-    <allow own="com.ubuntu.location.Service"/>
interfaces/builtin/location_control.go-    <allow send_destination="com.ubuntu.location.Service"/>
--
interfaces/builtin/location_observe.go-
interfaces/builtin/location_observe.go-var locationObservePermanentSlotDBus = []byte(`
interfaces/builtin/location_observe.go:<policy user="root">
interfaces/builtin/location_observe.go-    <allow own="com.ubuntu.location.Service"/>
interfaces/builtin/location_observe.go-    <allow own="com.ubuntu.location.Service.Session"/>
--
interfaces/builtin/modem_manager.go-</policy>
interfaces/builtin/modem_manager.go-
interfaces/builtin/modem_manager.go:<policy user="root">
interfaces/builtin/modem_manager.go-    <allow own="org.freedesktop.ModemManager1"/>
interfaces/builtin/modem_manager.go-</policy>
--
interfaces/builtin/network_manager.go-var networkManagerPermanentSlotDBus = []byte(`
interfaces/builtin/network_manager.go-<!-- DBus policy for NetworkManager (upstream version 1.2.2) -->
interfaces/builtin/network_manager.go:<policy user="root">
interfaces/builtin/network_manager.go-    <allow own="org.freedesktop.NetworkManager"/>
interfaces/builtin/network_manager.go-    <allow send_destination="org.freedesktop.NetworkManager"/>